packages/react-admin/src/auth/WithPermissions.js

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import compose from 'recompose/compose';
+import getContext from 'recompose/getContext';
+
+import { userCheck } from '../actions/authActions';
+import { AUTH_GET_PERMISSIONS } from '../auth/types';
+
+/**
+ * Retrieve permissions for a specific route.
+ *
+ * Useful for Route components ; used internally by CrudRoute.
```

Comment:
you should tell (and ullistrate in the example) that it injects the `permissions` prop to the child.

Reply:
Done